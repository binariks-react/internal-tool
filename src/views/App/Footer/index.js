import React from 'react';
import styled from 'styled-components';

const Wrapper = styled.div.attrs({ 'data-test': 'wrapper' })`
  display: flex;
  align-items: center;
  flex-shrink: 1;
  flex-grow: 1;
  flex-basis: auto;
  align-self: flex-end;
  justify-content: space-between;
  height: 70px;
  line-height: 70px;
  padding: 0 25px;
  border-top: 1px solid ${props => props.theme.colors.lightGrey};
`;

const PagesList = styled.ul.attrs({ 'data-test': 'pageList' })`
  display: flex;
  margin: 0;
  padding: 0;
`;

const PageOption = styled.li`
  font-family: Roboto, sans-serif;
  list-style: none;
  cursor: pointer;
  margin-right: 20px;
  color: ${props => props.theme.colors.primary};
  
  &:hover {
    color: ${props => props.theme.colors.text};
  }
`;

const Copyright = styled.span.attrs({ 'data-test': 'copyright' })`
  color: ${props => props.theme.colors.secondary};
`;

const Footer = props => (
  <Wrapper>
    <PagesList>
      <PageOption><span>About</span></PageOption>
      <PageOption><span>Services</span></PageOption>
      <PageOption><span>About</span></PageOption>
      <PageOption><span>Products</span></PageOption>
      <PageOption><span>Blog</span></PageOption>
    </PagesList>
    <Copyright>Copyright Â© 2019 Binariks</Copyright>
  </Wrapper>
);

export default Footer;
