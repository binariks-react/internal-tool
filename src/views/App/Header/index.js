import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Select from 'views/Components/Controls/Select';

const Wrapper = styled.div.attrs({ 'data-test': 'wrapper' })`
  display: flex;
  align-items: center;
  flex-shrink: 1;
  flex-grow: 1;
  flex-basis: auto;
  height: 70px;
  line-height: 70px;
  padding-left: 20px;
  box-shadow: ${props => props.theme.shadows.hover.dark};
`;

const Input = styled.input.attrs({ type: 'text', 'data-test': 'input' })`
  font-family: Roboto;
  font-size: 1rem;
  flex: 1;
  border: none;
  margin-left: 5px;
  height: 98%;
  outline: none;
`;

const NotificationWrapper = styled.div.attrs({ 'data-test': 'notifications' })`
  display: flex;
  align-items: center;
  height: 70px;
  position: relative;
  cursor: pointer;
  border-left: 1px solid ${props => props.theme.colors.lightGrey};
  padding: 0 25px;
`;

const SearchIcon = styled.i`
  color: ${props => props.theme.colors.icon};
`;

const NotificationIcon = styled.i`
  color: ${props => props.theme.colors.text};
`;

const NotificationsCount = styled.span.attrs({ 'data-test': 'notificationCount' })`
  position: absolute;
  width: 17px;
  height: 17px;
  border-radius: 50%;
  line-height: 17px;
  background-color: #c4183c;
  color: #ffffff;
  text-align: center;
  font-size: 12px;
  top: 35px;
  left: 40px;
  transition: all .3s;
`;

const UserSection = styled.div.attrs({ 'data-test': 'userSection' })`
  display: flex;
  align-items: center;
  margin-right: 10px;
  width: 215px;
  height: 100%;
  border-left: 1px solid ${props => props.theme.colors.lightGrey};
`;

const UserImage = styled.img`
  width: 42px;
  height: 42px;
  border-radius: 50%;
  margin: 0 15px 0 25px;
`;

const Header = (props) => {
  const { notifications } = props;
  const imgSrc = 'https://source.unsplash.com/random/50x50';
  const selectPlaceholder = 'Sierra Brooks';
  const selectOptions = [
    {
      value: 1,
      label: 'First',
    },
    {
      value: 2,
      label: 'Second',
    },
    {
      value: 3,
      label: 'Third',
    },
  ];

  return (
    <Wrapper>
      <SearchIcon className="material-icons">
        search
      </SearchIcon>
      <Input placeholder="Search for something..." />
      <NotificationWrapper>
        <NotificationIcon className="material-icons">
          notifications
        </NotificationIcon>
        {
          notifications && <NotificationsCount>{notifications}</NotificationsCount>
        }
      </NotificationWrapper>
      <UserSection>
        <UserImage src={`${imgSrc}`} />
        <Select
          options={selectOptions}
          placeholder={selectPlaceholder}
          style={{ width: '100%', height: '100%', border: 'none', outline: 'none', marginRight: '10px' }}
        />
      </UserSection>
    </Wrapper>
  );
};

Header.defaultProps = {
  notifications: null,
};

Header.propTypes = {
  notifications: PropTypes.number,
};

export default Header;
