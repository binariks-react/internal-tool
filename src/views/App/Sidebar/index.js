import React, { useState } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const getColor = (color, props) => (
  props ? props.theme.colors[color] : ({ theme: { colors } }) => colors[color]
);

const getActiveColor = (color, props) => (
  props ? props.theme.colors.activeColors[color] : ({ theme: { colors: { activeColors } } }) => activeColors[color]
);

const getHoverColors = (color, props) => (
  props ? props.theme.colors.activeColors[color] : ({ theme: { colors: { hoverColors } } }) => hoverColors[color]
);

const Wrapper = styled.div`
  width: 300px;
  display: block;
  height: 100vh;
  border-right: 1px solid ${getColor('lightGrey')};
`;

const Header = styled.div`
  width: 100%;
  height: 70px;
  border-bottom: 1px solid ${getColor('lightGrey')};
  text-align: center;
  line-height: 70px;
  color: ${getColor('text')};
`;

const Panel = styled.div`
  display: flex;
  align-items: center;
  width: 100%;
  height: 60px;
  border-left: ${({ selected, ...props }) => (
    selected ? `5px solid ${getColor('primary', props)}` : '5px solid transparent'
  )};
  border-bottom: 1px solid ${getColor('lightGrey')};
  padding-left: 30px;
  line-height: 60px;
  background-color: ${({ selected, ...props }) => selected ? getActiveColor('lightGrey', props) : getColor('white', props)};
  color: ${({ selected, ...props }) => selected ? getColor('primary', props) : getColor('text', props)};
  
  &:hover {
    background-color: ${getHoverColors('lightGrey')};
  }
`;

const Icon = styled.i`
  margin-right: 10px;
`;


const Sidebar = ({ titles }) => {
  const [selectedKey, setSelectedKey] = useState(titles.length && titles[0].title.replace(/\s/g, ''));

  return (
    <Wrapper className="wrapper">
      <Header>Dashboards</Header>
      {
        !!titles.length &&
        titles.map(({ title, icon }) => {
          const key = title.replace(/\s/g, '');
          return (
            <Panel
              key={key}
              selected={selectedKey === key}
              onClick={() => setSelectedKey(key)}
            >
              <Icon className="material-icons">
                { icon }
              </Icon>
              <span>{title}</span>
            </Panel>
          );
        })
      }
    </Wrapper>
  );
};

Sidebar.defaultProps = {
  titles: [],
};

Sidebar.propTypes = {
  titles: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    icon: PropTypes.string.isRequired,
  })),
};

export default Sidebar;
