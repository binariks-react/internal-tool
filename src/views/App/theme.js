export const getColor = (color, props) => (
  props ? props.theme.colors[color] : ({ theme }) => theme.colors[color]
);

export const getActiveColor = (color, props) => (
  props ? props.theme.colors.activeColors[color] : ({ theme }) => theme.colors.activeColors[color]
);

export const getHoverColor = (color, props) => (
  props ? props.theme.colors.hoverColors[color] : ({ theme }) => theme.colors.hoverColors[color]
);

export const getHoverShadowColor = (color, props) => (
  props ? props.theme.shadows.hover[color] : ({ theme }) => theme.shadows.hover[color]
);

export const getFocusShadowColor = (color, props) => (
  props ? props.theme.shadows.focus[color] : ({ theme }) => theme.shadows.focus[color]
);

export default {
  colors: {
    text: '#5a6169',
    lightGrey: '#e9ecef',
    darkGrey: '#becad6',
    border: '#e1e5eb',
    primary: '#007bff',
    secondary: '#5a6169',
    success: '#17c671',
    danger: '#c4183c',
    warning: '#ffb400',
    info: '#00b8d8',
    dark: '#212529',
    white: '#ffffff',
    royalBlue: '#674eec',
    java: '#1adba2',
    salmon: '#ff4169',
    hoverColors: {
      lightGrey: '#e9ecef',
      primary: '#006fe6',
      secondary: '#4e545b',
      success: '#14af64',
      danger: '#ad1535',
      warning: '#e6a200',
      info: '#00a2bf',
      dark: '#16181b',
      white: '#ffffff',
      royalBlue: '#5337ea',
      java: '#17c491',
      salmon: '#ff2855',
    },
    activeColors: {
      primary: '#0062cc',
      secondary: '#42484e',
      lightGrey: '#FBFBFB',
      success: '#129857',
      danger: '#97122e',
      warning: '#cc9000',
      info: '#008da5',
      dark: '#0a0c0d',
      white: '#007bff',
      royalBlue: '#3f20e7',
      java: '#15ad80',
      salmon: '#ff0e41',
    },
  },
  shadows: {
    hover: {
      primary: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(0,123,255,.25)',
      secondary: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(90,97,105,.25)',
      success: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(23,198,113,.25)',
      danger: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(196,24,60,.25)',
      warning: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(255,180,0,.25)',
      info: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(0,184,216,.25)',
      dark: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(33,37,41,.25)',
      white: '0 0.125rem 0.625rem rgba(129,142,163,.2), 0 0.0625rem 0.125rem rgba(129,142,164,.3)',
      royalBlue: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(103,78,236,.25)',
      java: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(26,219,162,.25)',
      salmon: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(255,65,105,.25)',
      inset: 'inset 0 3px 5px rgba(0,0,0,.125)!important',
    },
    focus: {
      primary: '0 0 0 3px rgba(0,123,255,.15),0 3px 15px rgba(0,123,255,.2),0 2px 5px rgba(0,0,0,.1)',
      secondary: '0 0 0 3px rgba(90,97,105,.15),0 3px 15px rgba(90,97,105,.2),0 2px 5px rgba(0,0,0,.1)',
      success: '0 0 0 3px rgba(23,198,113,.15),0 3px 15px rgba(23,198,113,.2),0 2px 5px rgba(0,0,0,.1)',
      danger: '0 0 0 3px rgba(196,24,60,.15),0 3px 15px rgba(196,24,60,.2),0 2px 5px rgba(0,0,0,.1)',
      warning: '0 0 0 3px rgba(255,180,0,.15),0 3px 15px rgba(255,180,0,.2),0 2px 5px rgba(0,0,0,.1)',
      info: '0 0 0 3px rgba(0,184,216,.15),0 3px 15px rgba(0,184,216,.2),0 2px 5px rgba(0,0,0,.1)',
      dark: '0 0 0 3px rgba(33,37,41,.15),0 3px 15px rgba(33,37,41,.2),0 2px 5px rgba(0,0,0,.1)',
      white: '0 0 0 3px rgba(251,251,251,.15),0 3px 15px rgba(251,251,251,.2),0 2px 5px rgba(0,0,0,.1)',
      royalBlue: '0 0 0 3px rgba(103,78,236,.15),0 3px 15px rgba(103,78,236,.2),0 2px 5px rgba(0,0,0,.1)',
      java: '0 0 0 3px rgba(26,219,162,.15),0 3px 15px rgba(26,219,162,.2),0 2px 5px rgba(0,0,0,.1)',
      salmon: '0 0 0 3px rgba(255,65,105,.15),0 3px 15px rgba(255,65,105,.2),0 2px 5px rgba(0,0,0,.1)',
      switch: '0 3px 8px 1px rgba(0,0,0,.2)',
    },
  },
};
