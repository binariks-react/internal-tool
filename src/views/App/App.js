import React from 'react';
import styled, { ThemeProvider } from 'styled-components';
import Sidebar from 'views/App/Sidebar/Sidebar';
import Header from 'views/App/Header/Header';
import Footer from 'views/App/Footer/Footer';
import theme from './theme';
import Routes from './Routes/Routes';

const Wrapper = styled.div`
  display: flex;

  button + button {
    margin-left: 20px;
  }
`;

const MainSection = styled.main`
  width: 100%;
  height: 100vh;
  overflow-y: auto;
`;


const titles = [
  {
    title: 'Blog Dashboard',
    icon: 'edit',
  },
  {
    title: 'Blog Posts',
    icon: 'vertical_split',
  },
  {
    title: 'Add New Post',
    icon: 'note_add',
  },
  {
    title: 'Forms & Components',
    icon: 'view_module',
  },
  {
    title: 'Tables',
    icon: 'table_chart',
  },
  {
    title: 'User Profile',
    icon: 'person',
  },
  {
    title: 'Errors',
    icon: 'error',
    route: '/error',
  },
];

const App = () => (
  <div className="App">
    <ThemeProvider theme={theme}>
      <Wrapper>
        <Sidebar titles={titles} />
        <MainSection>
          <Header notifications={12} />
          <Routes />
          <Footer />
        </MainSection>
      </Wrapper>
    </ThemeProvider>
  </div>
);

export default App;
