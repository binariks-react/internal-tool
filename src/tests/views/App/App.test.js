import React from 'react';
import { shallow } from 'enzyme';
import App from 'views/App/App';

describe('App', () => {
  it('snapshot', () => {
    const app = shallow(
      <App />
    );
    expect(app).toMatchSnapshot();
  });
});
