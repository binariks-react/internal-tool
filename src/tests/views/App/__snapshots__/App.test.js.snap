// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<div
  className="App"
>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "activeColors": Object {
            "danger": "#97122e",
            "dark": "#0a0c0d",
            "info": "#008da5",
            "java": "#15ad80",
            "primary": "#0062cc",
            "royalBlue": "#3f20e7",
            "salmon": "#ff0e41",
            "secondary": "#42484e",
            "success": "#129857",
            "warning": "#cc9000",
            "white": "#007bff",
          },
          "border": "#e1e5eb",
          "danger": "#c4183c",
          "dark": "#212529",
          "darkGrey": "#becad6",
          "hoverColors": Object {
            "border": "#b3bdcc",
            "danger": "#ad1535",
            "dark": "#16181b",
            "info": "#00a2bf",
            "java": "#17c491",
            "primary": "#006fe6",
            "royalBlue": "#5337ea",
            "salmon": "#ff2855",
            "secondary": "#4e545b",
            "success": "#14af64",
            "warning": "#e6a200",
            "white": "#ffffff",
          },
          "icon": "#abb6bf",
          "info": "#00b8d8",
          "java": "#1adba2",
          "lightGrey": "#e9ecef",
          "primary": "#007bff",
          "royalBlue": "#674eec",
          "salmon": "#ff4169",
          "secondary": "#5a6169",
          "success": "#17c671",
          "text": "#5a6169",
          "warning": "#ffb400",
          "white": "#ffffff",
        },
        "shadows": Object {
          "focus": Object {
            "danger": "0 0 0 3px rgba(196,24,60,.15),0 3px 15px rgba(196,24,60,.2),0 2px 5px rgba(0,0,0,.1)",
            "dark": "0 0 0 3px rgba(33,37,41,.15),0 3px 15px rgba(33,37,41,.2),0 2px 5px rgba(0,0,0,.1)",
            "info": "0 0 0 3px rgba(0,184,216,.15),0 3px 15px rgba(0,184,216,.2),0 2px 5px rgba(0,0,0,.1)",
            "java": "0 0 0 3px rgba(26,219,162,.15),0 3px 15px rgba(26,219,162,.2),0 2px 5px rgba(0,0,0,.1)",
            "primary": "0 0 0 3px rgba(0,123,255,.15),0 3px 15px rgba(0,123,255,.2),0 2px 5px rgba(0,0,0,.1)",
            "royalBlue": "0 0 0 3px rgba(103,78,236,.15),0 3px 15px rgba(103,78,236,.2),0 2px 5px rgba(0,0,0,.1)",
            "salmon": "0 0 0 3px rgba(255,65,105,.15),0 3px 15px rgba(255,65,105,.2),0 2px 5px rgba(0,0,0,.1)",
            "secondary": "0 0 0 3px rgba(90,97,105,.15),0 3px 15px rgba(90,97,105,.2),0 2px 5px rgba(0,0,0,.1)",
            "success": "0 0 0 3px rgba(23,198,113,.15),0 3px 15px rgba(23,198,113,.2),0 2px 5px rgba(0,0,0,.1)",
            "switch": "0 3px 8px 1px rgba(0,0,0,.2)",
            "warning": "0 0 0 3px rgba(255,180,0,.15),0 3px 15px rgba(255,180,0,.2),0 2px 5px rgba(0,0,0,.1)",
            "white": "0 0 0 3px rgba(251,251,251,.15),0 3px 15px rgba(251,251,251,.2),0 2px 5px rgba(0,0,0,.1)",
          },
          "hover": Object {
            "danger": "0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(196,24,60,.25)",
            "dark": "0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(33,37,41,.25)",
            "info": "0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(0,184,216,.25)",
            "inset": "inset 0 3px 5px rgba(0,0,0,.125)!important",
            "java": "0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(26,219,162,.25)",
            "primary": "0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(0,123,255,.25)",
            "royalBlue": "0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(103,78,236,.25)",
            "salmon": "0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(255,65,105,.25)",
            "secondary": "0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(90,97,105,.25)",
            "success": "0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(23,198,113,.25)",
            "warning": "0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(255,180,0,.25)",
            "white": "0 0.125rem 0.625rem rgba(129,142,163,.2), 0 0.0625rem 0.125rem rgba(129,142,164,.3)",
          },
        },
      }
    }
  >
    <Routes />
  </ThemeProvider>
</div>
`;
